sql/table_iter.go

```diff
@@ -0,0 +1,75 @@
+package sql
+
+import (
+	"context"
+	"io"
+)
+
+// TableRowIter is an iterator over the partitions in a table.
+type TableRowIter struct {
+	ctx        *Context
```

Comment:
Sadness but this seems idiomatic for now... Better would be `Next(ctx sql.Context)` I think...

Reply:
Maybe someday. A lot of these interfaces should have context but don't.